lib/chartjs/chart_helpers.rb

```diff
@@ -49,16 +53,9 @@ def chart(klass, data, options)
 
             #{legend if generate_legend}
           };
-          /* W3C standard */
-          if (window.addEventListener) {
-            window.addEventListener("load", initChart, false);
-            document.addEventListener("page:load", initChart, false);
-          }
-          /* IE */
-          else if (window.attachEvent) {
-            window.attachEvent("onload", initChart);
-            document.attachEvent("page:load", initChart);
-          }
+          
+          initChart();
```

Comment:
OK, I understand the situation now, thanks.

Wouldn't your `initChart()` be called as soon as the block of javascript is written out?  Which could be before the page load event, and thus run into the race condition mentioned in the Chart.js docs?